tests/test_app_upload_jurisdictions_file.py

```diff
@@ -0,0 +1,179 @@
+    election_id = json.loads(rv.data)['electionId']
+
+    rv = client.post(
+        f'/election/{election_id}/jurisdictions_file',
+        data={'jurisdictions': (io.BytesIO(b'Jurisdiction,Admin Email'), 'jurisdictions.csv')})
+    assert rv.status_code == 200
+    assert json.loads(rv.data) == {'status': 'ok'}
+
+    election = Election.query.filter_by(id=election_id).one()
+    assert election.jurisdictions == []
+    assert JurisdictionAdministration.query.count() == 0
+    assert User.query.count() == 0
+
+
+def test_single_jurisdiction_single_admin(client):
+    rv = post_json(client, '/election/new', {})
+    election_id = json.loads(rv.data)['electionId']
+
+    rv = client.post(f'/election/{election_id}/jurisdictions_file',
+                     data={
+                         'jurisdictions':
+                         (io.BytesIO(b'Jurisdiction,Admin Email\nJ1,a1@example.com'),
+                          'jurisdictions.csv')
+                     })
+    assert rv.status_code == 200
+    assert json.loads(rv.data) == {'status': 'ok'}
+
+    election = Election.query.filter_by(id=election_id).one()
+    assert [j.name for j in election.jurisdictions] == ['J1']
+
+    jurisdiction = election.jurisdictions[0]
+    assert [a.user.email for a in jurisdiction.jurisdiction_administrations] == ['a1@example.com']
+
+
+def test_single_jurisdiction_multiple_admins(client):
+    rv = post_json(client, '/election/new', {})
+    election_id = json.loads(rv.data)['electionId']
+
+    rv = client.post(
+        f'/election/{election_id}/jurisdictions_file',
+        data={
+            'jurisdictions':
+            (io.BytesIO(b'Jurisdiction,Admin Email\nJ1,a1@example.com\nJ1,a2@example.com'),
+             'jurisdictions.csv')
+        })
+    assert rv.status_code == 200
+    assert json.loads(rv.data) == {'status': 'ok'}
+
+    election = Election.query.filter_by(id=election_id).one()
+    assert [j.name for j in election.jurisdictions] == ['J1']
+
+    jurisdiction = election.jurisdictions[0]
+    assert [a.user.email for a in jurisdiction.jurisdiction_administrations
+            ] == ['a1@example.com', 'a2@example.com']
+
+
+def test_multiple_jurisdictions_single_admin(client):
+    rv = post_json(client, '/election/new', {})
+    election_id = json.loads(rv.data)['electionId']
+
+    rv = client.post(
+        f'/election/{election_id}/jurisdictions_file',
+        data={
+            'jurisdictions':
+            (io.BytesIO(b'Jurisdiction,Admin Email\nJ1,a1@example.com\nJ2,a1@example.com'),
+             'jurisdictions.csv')
+        })
+    assert rv.status_code == 200
+    assert json.loads(rv.data) == {'status': 'ok'}
+
+    election = Election.query.filter_by(id=election_id).one()
+    assert [j.name for j in election.jurisdictions] == ['J1', 'J2']
+
+    for jurisdiction in election.jurisdictions:
+        assert [a.user.email
+                for a in jurisdiction.jurisdiction_administrations] == ['a1@example.com']
```

Comment:
This is officially the best tested feature in the codebase.

Reply:
😂